### PR TITLE
toolong: init at 1.4.0

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

Fixes:
-  #327706

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/toolong/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  passthru.tests.version = testers.testVersion {
+    package = toolong;
+    command = "${lib.getExe toolong} --version";
+  };
+
+  meta = with lib; {
+    description = "Terminal application to view, tail, merge, and search log files (plus JSONL)";
+    license = licenses.mit;
+    homepage = "www.textualize.io";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/toolong/default.nix:45]:
```suggestion
    homepage = "https://www.textualize.io";
```

**Review by Aleksanaa — COMMENTED** [review]
If it's not used as a python module, just put it in pkgs/by-name. Otherwise you can add `toolong = with python3Packages; toPythonApplication toolong` in `all-packages.nix`.

File: pkgs/by-name/to/toolong/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  passthru.tests.version = testers.testVersion {
+    package = toolong;
+    command = "${lib.getExe toolong} --version";
+  };
+
+  meta = with lib; {
+    description = "Terminal application to view, tail, merge, and search log files (plus JSONL)";
+    license = licenses.mit;
+    homepage = "https://www.textualize.io";
```
Comment (natsukium) [inline pkgs/by-name/to/toolong/package.nix:41]:
```suggestion
    homepage = "https://github.com/textualize/toolong";
```
https://github.com/Textualize/toolong/blob/5aa22ee878026f46d4d265905c4e1df4d37842ae/pyproject.toml#L9

This one would be more suitable as an actual `toolong` homepage.

**Review by natsukium — APPROVED** [review]
lgtm, thanks

**Comment by natsukium** [comment]
@ofborg build toolong

**Comment by gador** [comment]
ofborg complains about 
```
error: tree-sitter0_21-0.21.3 not supported for interpreter python3.12
```
and `nixpkgs-review pr 327803` reports `Nothing to build`

File: pkgs/by-name/to/toolong/package.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  python3Packages,
```
Comment (SuperSandro2000) [inline pkgs/by-name/to/toolong/package.nix:3]:
```suggestion
  python311Packages,
```
we probably want to use 3.11 until treesitter works with 3.12

**Comment by Sigmanificient** [comment]
Result of `nixpkgs-review pr 327803` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>toolong</li>
    <li>toolong.dist</li>
  </ul>
</details>

**Pull request merged**